### ISSUE TITLE
Sling MPD Live Issue

### ISSUE THREAD

**Issue by eracknaphobia** [peak3d/inputstream.adaptive#880]
While trying to debug an [issue](https://github.com/d21spike/plugin.video.sling/issues/40) in the Sling TV Addon it was [suggested](https://github.com/d21spike/plugin.video.sling/issues/40#issuecomment-1017708480) to create an issue here.

1. First segment played `https://p-cdn4-e-cg14-linear-cbd46b77.movetv.com/15803/live/ESPNHD/4a8a1a5479cd11ecbc620025b5472112/video/vid06/0000493d.m4s`

2. MPD file
[spanning_ads-real-pto.mpd.txt](https://github.com/xbmc/inputstream.adaptive/files/7908021/spanning_ads-real-pto.mpd.txt)

3. Epoch  when fired `1642706751`

```
wall clock time = 2022-01-20T19:25:51Z (1642706751)
availabilityStartTime = 2022-01-20T19:00:00Z (1642705200)
duration = 2048
timescale = 1000
startNumber = 18015

((wall clock time - availabilityStartTime ) / (duration / timescale )) + startNumber

((1642706751 - 1642705200) / (2048 / 1000)) + 18015 
(1511 / 2.048) + 18015 
737 + 18015 = 18752
which is only 3 segments off of the first segment played in the browser (493d = 18749)
```

Hope this is helpful...

**Comment by matthuisman** [comment]
@eracknaphobia IA is doing a few things wrong
1) https://github.com/xbmc/inputstream.adaptive/blob/Nexus/src/parser/DASHTree.cpp#L1285
 tpl.timescale / tpl.duration should be  tpl.duration/ tpl.timescale 
2) overallSeconds shouldnt be remove here. 
So the formula should be:
`seg.range_end_ += (static_cast<int64_t>(dash->stream_start_ - dash->available_time_ - sample_time)) * tpl.duration / tpl.timescale + 1;`
sample_time = the start time of current period = correct.
eg. if we are on 2nd period and it started 10s after the 1st period - we need to remove 10s

3) We are "double dipping" i think
  Here: https://github.com/xbmc/inputstream.adaptive/blob/Nexus/src/parser/DASHTree.cpp#L1242
  we are calculating the total number of segments based off oversalseconds (which will be based off the period duration = wrong in our case as its live and the period isnt finished) 
    This number of segements gets added here: https://github.com/xbmc/inputstream.adaptive/blob/Nexus/src/parser/DASHTree.cpp#L1300
     But we also add in our above formula as well!

Wait, just noticed we doing *, so the reverse of duration and timescale may be ok

also your maths was a bit off

```
wall clock time = 2022-01-20T19:25:51Z (1642706751)
availabilityStartTime = 2022-01-20T19:00:15Z (1642705215)
duration = 2048
timescale = 1000
startNumber = 18015

((wall clock time - availabilityStartTime ) / (duration / timescale )) + startNumber

((1642706751 - 1642705215) / (2048 / 1000)) + 18015 
(1,536 / 2.048) + 18015 
750 + 18015 = 18765 
= 494D
= 16 segments in front of browser = 32.76s 

This could be simply the buffer.
IA will start 12s behind our final segment I believe.
```

**Comment by matthuisman** [comment]
@eracknaphobia 
please try Matrix build from here:
https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-881/5/artifacts

**Comment by eracknaphobia** [comment]
@matthuisman  with the latest build live channels are working as expected. 

NBA streams start from beginning, instead of live. 

Adding the properties you mentioned seems to fix the NBA streams, I will have to do more testing.
```
liz.setProperty('ResumeTime', '43200')
liz.setProperty('TotalTime', '1')
```

**Comment by matthuisman** [comment]
When it starts from start, can you download that manifest and upload it.
I assume it hill have multiple periods.
One short 3s advert probably.
IA should be starting playback at the end of that period. As it's short, it probably seems like it's the start of the stream

also, what does Kodi show as the length of the timeline?
ie. how far back and forward can you rwd / fwd?
any little lines on the timeline (these are periods)

**Comment by eracknaphobia** [comment]
Live games won't start until 7pm EST tonight. I'll hop on then and get you the info requested.

**Comment by eracknaphobia** [comment]
Here's the manifest file.
[spanning_ads-3.mpd.txt](https://github.com/xbmc/inputstream.adaptive/files/7917269/spanning_ads-3.mpd.txt)

The time looks right in the timeline, the game should be about 20 minutes in but the video feed starts at the very beginning of the broadcast.
![image](https://user-images.githubusercontent.com/8268725/150616511-d9d856dd-0b0c-4478-857b-5b5bd813e5a9.png)

**Comment by matthuisman** [comment]
As I thought, that has a very short 2s ,1st period. So seems it is working fine. If you use the resume from 'hack" everything works as expected? All other streams also work fine?

**Comment by eracknaphobia** [comment]
Yep, setting ResumeTime fixes these streams and doesn't seem to effect the other streams.

**Comment by matthuisman** [comment]
As long as it's only set for live streams, you'll be fine

**Comment by matthuisman** [comment]
The test builds are here
https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-881/5/artifacts

**Comment by tmm1** [comment]
Might be fixed by https://github.com/xbmc/inputstream.adaptive/pull/1176

**Comment by CastagnaIT** [comment]
wrong downloaded segment should be fixed by #1508
if you have possibility give a feedback
test builds: https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-1508/8/artifacts

**Comment by CastagnaIT** [comment]
closed as no replies from issue author
